src/test/java/org/opensearch/security/test/helper/rest/RestHelper.java

```diff
@@ -121,6 +127,25 @@ public String executeSimpleRequest(final String request) throws Exception {
 		}
 	}
 
+	public HttpResponse[] executeMultipleAsyncPutRequest(final int numOfRequests, final String request, String body) throws Exception {
+		final ExecutorService executorService = Executors.newFixedThreadPool(numOfRequests);
+		Future<HttpResponse>[] futures = new Future[numOfRequests];
+		for (int i = 0; i < numOfRequests; i++) {
+			futures[i] = executorService.submit(() -> executePutRequest(request, body, new Header[0]));
+		}
+		executorService.shutdown();
+		try {
+			if (!executorService.awaitTermination(1, TimeUnit.SECONDS)) {
```

Comment:
There is no need to await termination as we call `future.get()` in `HttpResponse::from`